core/src/main/java/org/apache/calcite/rel/metadata/RelMdUtil.java

```diff
@@ -943,9 +943,9 @@ private static boolean isNonNegative(Double result, boolean fail) {
    * Removes cached metadata values for specified RelNode.
    *
    * @param rel RelNode whose cached metadata should be removed
+   * @return true if cache for the provided RelNode was not empty
    */
-  public static void clearCache(RelNode rel) {
-    rel.getCluster().getMetadataQuery().clearCache(rel);
+  public static boolean clearCache(RelNode rel) {
```

Comment:
I am not sure what is the standard procedure of changing a public API.

Reply:
I think void to boolean is fine.

Reply:
Agree with @hsyuan . It will not break anything.

Reply:
Thanks a lot for your clarification.

Reply:
I'm wondering how to give a valid test, e.g. even though we clear the cache that should not do, the tests still passes.

Reply:
I tried to add a test case. But it is really difficult. We can see the problem is just there.